src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -162,41 +164,63 @@ IFACEMETHODIMP ZoneSet::AddZone(winrt::com_ptr<IZone> zone) noexcept
     return S_OK;
 }
 
-IFACEMETHODIMP_(winrt::com_ptr<IZone>)
-ZoneSet::ZoneFromPoint(POINT pt) noexcept
+IFACEMETHODIMP_(std::vector<int>)
+ZoneSet::ZonesFromPoint(POINT pt) noexcept
 {
-    winrt::com_ptr<IZone> smallestKnownZone = nullptr;
-    // To reduce redundant calculations, we will store the last known zones area.
-    int smallestKnownZoneArea = INT32_MAX;
-    for (auto iter = m_zones.rbegin(); iter != m_zones.rend(); iter++)
+    const int SENSITIVITY_RADIUS = 20;
+    std::vector<int> capturedZones;
+    for (size_t i = 0; i < m_zones.size(); i++)
+    {
+        auto zone = m_zones[i];
+        RECT newZoneRect = zone->GetZoneRect();
+        if (newZoneRect.left - SENSITIVITY_RADIUS <= pt.x && pt.x <= newZoneRect.right + SENSITIVITY_RADIUS &&
+            newZoneRect.top - SENSITIVITY_RADIUS <= pt.y && pt.y <= newZoneRect.bottom + SENSITIVITY_RADIUS &&
+            newZoneRect.left < newZoneRect.right && newZoneRect.top < newZoneRect.bottom) // proper zone
+        {
+            capturedZones.emplace_back(i);
```

Comment:
let's not break the build!
```suggestion
            capturedZones.emplace_back(static_cast<int>(i));
```